components/script/dom/blob.rs

```diff
@@ -26,7 +26,7 @@ use uuid::Uuid;
 #[derive(JSTraceable)]
 pub struct FileBlob {
     id: SelectedFileId,
-    name: PathBuf,
+    name: Option<PathBuf>,
```

Comment:
This is critical -- The "File" might not be the real file, the promoted ones just don't have a filename.